service/raft/raft_group0.cc

```diff
@@ -524,6 +526,13 @@ future<> raft_group0::setup_group0(db::system_keyspace& sys_ks, const std::unord
         co_return;
     }
 
+    if (replace_info && !replace_info->raft_id) {
+        auto msg = format("Cannot perform replace operation: Raft ID of the replaced server (IP: {}) is missing.",
+                          replace_info->ip_addr);
+        group0_log.error("{}", msg);
+        throw std::runtime_error{std::move(msg)};
```

Comment:
Why log+throw? It will be logged twice.

Reply:
The second log will happen out of context, after the replace procedure fails, after all services have shut down, so there will be all this logging related to shutting down services before my failure.

I'd like the log message to be printed in-context, so I can read the logs sequentially and see: replace started, this happened, that happened, and then there was this failure. Hence the `group0_log.error(...)` call.

And I don't think there's any advantage to throwing an empty exception just to avoid the second log, in fact I think the second log is also useful: it tells us the reason why Scylla decided to shutdown (and then you can browse the logs to see in what context the failure happened).

Reply:
Fair enough. 

Although, the same argument can be made about any exception which can trigger shutdown. What should be the systematic approach to this? Maybe every subsystem should catch exceptions around its startup at the highest level, and log it there before cleanup starts?

Reply:
> Maybe every subsystem should catch exceptions around its startup at the highest level, and log it there before cleanup starts?

Maybe, but there may be multiple steps to initializing a subsystem, for example `raft_group0` has `setup_group0` and then `finish_setup_after_join` - which, if it fails, will also fail startup.

Reply:
For such important logs, it is not a big deal we see it more than once in the log. It is much worse not seeing it.